src/Datadog.Trace/HttpOverStreams/HttpContent/StreamContent.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Datadog.Trace.HttpOverStreams.HttpContent
+{
+    internal class StreamContent : IHttpContent
+    {
+        private const int BufferSize = 10240;
+
+        public StreamContent(Stream stream, long? length)
+        {
+            Stream = stream;
+            Length = length;
+        }
+
+        public Stream Stream { get; }
+
+        public long? Length { get; }
+
+        public void CopyTo(Stream destination, int count)
+        {
+            var bytes = new byte[Math.Min(count, BufferSize)];
+            int bytesLeft = count;
+
+            while (bytesLeft > 0)
+            {
+                int bytesRead = Stream.Read(bytes, 0, bytes.Length);
+                destination.Write(bytes, 0, bytesRead);
```

Comment:
Can you call `Stream.CopyTo` instead of managing the bytes yourself?

Reply:
We want to read only the specified number of bytes (`count` parameter), and `Stream.CopyTo` does not provide this.

Reply:
I think `CopyToAsync` is doing it wrong. I don't know why this method has a `count` parameter and `CopyToAsync` doesn't. Probably leftover from my POC (and I was only using one of these and not both).

Reply:
Sorry, I should have been more specific when I said "I think `CopyToAsync` below is doing it wrong."

The `StreamContent.CopyToAsync(Stream)` overload below calls `Stream.CopyToAsync(Stream, int)`. However, the second parameter is the buffer size ([docs](https://docs.microsoft.com/en-us/dotnet/api/system.io.stream.copytoasync?view=net-5.0#System_IO_Stream_CopyToAsync_System_IO_Stream_System_Int32_)), not the number of bytes to copy, so this line looks like a bug to me:
```csharp
await Stream.CopyToAsync(destination, (int)Length).ConfigureAwait(false);
```

Reply:
I am using the method provided by this class now.

Reply:
I actually am using the `Stream.CopyToAsync` method again, but with a max buffer limit.
For now the deserialization will fail if the response is too big, but there is a reasonable limit on what is copied in memory.